From 946a82d6042097a0b3c5e072df75da9ffcbbc44c Mon Sep 17 00:00:00 2001
From: Ajay Kumar Gupta <ajay.gupta@ti.com>
Date: Thu, 24 May 2012 15:45:45 +0530
Subject: [PATCH 12/18] usb: musb: cppi41: use dsb() to make sure PDs are
 updated to memory

Before pushing the tx/rx cppi packet descriptors to respective input queues
of TX/RX DMA channels, use memory barrier dsb() to make sure the descriptors
contents are written to memory

Signed-off-by: Ravi B <ravibabu@ti.com>
---
 drivers/usb/musb/cppi41_dma.c |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/musb/cppi41_dma.c b/drivers/usb/musb/cppi41_dma.c
index 3665d0f..bf74cf3 100644
--- a/drivers/usb/musb/cppi41_dma.c
+++ b/drivers/usb/musb/cppi41_dma.c
@@ -653,6 +653,9 @@ static unsigned cppi41_next_tx_segment(struct cppi41_channel *tx_ch)
 		dev_dbg(musb->controller, "TX PD %p: buf %08x, len %08x, pkt info %08x\n", curr_pd,
 		    hw_desc->buf_ptr, hw_desc->buf_len, hw_desc->pkt_info);
 
+		/* make sure descriptor details are updated to memory*/
+		dsb();
+
 		/* enable tx fifo empty interrupt */
 		if (tx_ch->txfifo_intr_enable) {
 			dev_dbg(musb->controller, "txs(%p %d) enable txFintr\n",
@@ -869,6 +872,10 @@ static unsigned cppi41_next_rx_segment(struct cppi41_channel *rx_ch)
 
 		if (en_bd_intr)
 			hw_desc->orig_buf_len |= CPPI41_PKT_INTR_FLAG;
+
+		/* make sure descriptor details are updated to memory*/
+		dsb();
+
 		/*
 		 * Push the free Rx packet descriptor
 		 * to the free descriptor/buffer queue.
-- 
1.7.7.6

